### PR TITLE
Fix some broken doc references

### PR DESCRIPTION
**Pull request by tvolkert**
## Description

This begins to fix our broken doc references.  There's too many breakages to fix outright, so I'm gonna bail on the direct fix and send a PR to add an allowed list for the existing failures, so we can drive that list to zero and keep from making things worse.  In the meantime, this PR makes things slightly better.

## Related Issues

https://github.com/flutter/flutter/issues/31931

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*

### CODE REVIEW

**Review by HansMuller — APPROVED** [review]
LGTM. The changes in `slider_theme.dart` are substantial, @JoseAlba and/or @clocksmith should have a look at them.

**Comment by tvolkert** [comment]
> The changes in slider_theme.dart are substantial

Yeah, they were declaring dartdoc templates at the top of the file, which doesn't work -- it ends up inlining those templates into the class doc, which isn't what was intended.

**Comment by tvolkert** [comment]
The test failures are real - with this change, we're (correctly) exporting `src/material/curves.dart`, but the symbols therein are colliding with the `animations` package.

**Comment by HansMuller** [comment]
@goderbauer - you should take a look at https://github.com/flutter/flutter/pull/62317#issuecomment-664463328

**Comment by tvolkert** [comment]
The conflict comes from:

* https://github.com/flutter/packages/blob/master/packages/animations/lib/src/utils/curves.dart
* https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/curves.dart

**Comment by goderbauer** [comment]
We should just remove the copy of Curves from the animations package and use the definition rom material: https://github.com/flutter/packages/pull/182

**Comment by goderbauer** [comment]
Thanks for doing this clean-up by the way. I started some doc cleanup as well in https://github.com/flutter/flutter/pull/62167 and https://github.com/flutter/flutter/pull/62071. I was gonna look at some of the broken references next.

BTW, I was wondering if we could clean up the references one folder at a time (e.g. `material`, `widget`, etc.) and then place a `dartdoc_options.yaml` in that folder that elevates the (ignored) warning of a broken doc references to an actual error.

**Review by clocksmith — APPROVED** [review]

File: packages/flutter/lib/material.dart [hunk 1]
```diff
@@ -49,6 +49,7 @@ export 'src/material/circle_avatar.dart';
 export 'src/material/color_scheme.dart';
 export 'src/material/colors.dart';
 export 'src/material/constants.dart';
+export 'src/material/curves.dart';
```
Comment (goderbauer) [inline packages/flutter/lib/material.dart:52]: Can we do this one in a separate PR to make a potential revert easier? This one is a little more risky and may break things (see animations package).

Comment (tvolkert) [reply packages/flutter/lib/material.dart:52]: ack - will update it tomorrow morning

Comment (tvolkert) [reply packages/flutter/lib/material.dart]: Done

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by goderbauer** [comment]
Looks like this one needs to be rebased ontop the latest master to pick up the version resolvment fix to make cirrus happy.
